src/odemis/driver/technolution.py

```diff
@@ -323,27 +333,26 @@ def system_checks(self):
         except Exception:
             logging.exception("Performing system checks failed. Could not perform a successful call to %s ." % item_name)
 
-    def checkMegaFieldExists(self, mega_field_id, storage_dir):
+    def checkMegaFieldExists(self, file_name, path):
         """
-        Check if filename complies with set allowed characters.
-        :param mega_field_id (string): name of the mega field.
-        :param storage_dir (string): path to the mega field.
-        :return (bool): True if mega field exists.
+        Checks if a megafield image already exists on the external storage for a given path and file name
+        (megafield id).
+        :param file_name: (str) Name of the mega field (id).
+        :param path: (str) Path to the mega field.
+        :return: (bool) True if the mega field exists on external storage. False if it does not exist yet.
+        :raises: (ValueError) When the input arguments contain invalid characters.
         """
-        ASM_FILE_ILLEGAL_CHARS = r'[^a-z0-9_()-]'
-        ASM_PATH_ILLEGAL_CHARS = r'[^A-Za-z0-9/_()-]'
-        if re.search(ASM_PATH_ILLEGAL_CHARS, storage_dir):
-            logging.error("The specified storage directory contains invalid characters, cannot check if mega field "
-                          "exists (only the characters '%s' are allowed)." % ASM_FILE_ILLEGAL_CHARS[2:-1])
-            return False
+        if re.search(ASM_PATH_CHARS, path):
```

Comment:
The `ASM_PATH_CHARS` variable actually looks for characters which are not allowed. That's what the "^" is for. That's why the code raises an error in case there is a match. So the previous names with "ILLEGAL" were clearer. That said, I think it'd be even clearer to change the logic  by checking that *every* character is legal. You could write it like this:
```python
ASM_PATH_CHARS = r'[A-Za-z0-9/_()-]*'
ASM_FILE_CHARS = r'[A-Za-z0-9_()-]+' # + instead of * to force at least one character
if not re.fullmatch(ASM_PATH_CHARS, path):
    raise ...
:
```